apps/condo/domains/user/schema/ConfirmPhoneAction.js

```diff
@@ -0,0 +1,245 @@
+            },
+        },
+        token: {
+            schemaDoc: 'Unique token to complete confirmation',
+            type: Text,
+            isUnique: true,
+            isRequired: true,
+        },
+        smsCode: {
+            schemaDoc: 'Last sms code sent to user',
+            type: Integer,
+            length: SMS_CODE_LENGTH,
+        },
+        smsCodeRequestedAt: {
+            schemaDoc: 'Time when sms code was requested',
+            type: DateTimeUtc,
+            isRequired: true,
+        },
+        smsCodeExpiresAt: {
+            schemaDoc: 'Time when sms code becomes not valid',
+            type: DateTimeUtc,
+            isRequired: true,
+        },
+        retries: {
+            schemaDoc: 'Number of times sms code input from user failed',
+            type: Integer,
+            defaultValue: 0,
+        },
+        isPhoneVerified: {
+            schemaDoc: 'Phone verification flag. User verify phone by access to secret sms message',
+            type: Checkbox,
+            defaultValue: false,
+        },
+        requestedAt: {
+            schemaDoc: 'DateTime when confirm phone action was started',
+            type: DateTimeUtc,
+            isRequired: true,
+        },
+        expiresAt: {
+            schemaDoc: 'When confirm phone action becomes invalid',
+            type: DateTimeUtc,
+            isRequired: true,
+        },
+        completedAt: {
+            schemaDoc: 'When confirm phone action was completed',
+            type: DateTimeUtc,
+            isRequired: false,
+        },
+
+    },
+    plugins: [uuided(), softDeleted(), historical()],
+    access: {
+        read: access.canReadConfirmPhoneActions,
+        create: access.canManageConfirmPhoneActions,
+        update: access.canManageConfirmPhoneActions,
+        delete: false,
+        auth: true,
+    },
+})
+
+
+const ConfirmPhoneActionService = new GQLCustomSchema('ConfirmPhoneActionService', {
+    queries: [
+        { 
+            access: true,
+            schema: 'getPhoneByConfirmPhoneActionToken(token: String!): String',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { token } = args
+                const now = extra.extraNow || Date.now()
+                const actions = await ConfirmPhoneActionGQL.getAll(context.createContext({ skipAccessControl: true }), { 
+                    token,
+                    expiresAt_gte: new Date(now).toISOString(),
+                    completedAt: null,
+                })
+                if (isEmpty(actions)) {
+                    throw new Error(`${CONFIRM_PHONE_ACTION_EXPIRED}]: Unable to find confirm phone action by token`)
+                }
+                const { phone } = actions[0]
+                return phone
+            },
+        },
+    ],    
+    mutations: [
+        {
+            access: true,
+            schema: 'startConfirmPhoneAction(phone: String!, dv:Int!, sender: JSON!): String',
```

Comment:
it's better to use `startConfirmPhoneAction(data: StartConfirmPhoneActionInput!): StartConfirmPhoneActionOutput` to extend the return data and mutation arguments